src/ui/common/src/components/workflows/StatusBar.tsx

```diff
@@ -124,9 +124,10 @@ const ActiveWorkflowStatusTab: React.FC<ActiveWorkflowStatusTabProps> = ({
       }}
     >
       {listItems.map((listItem, index) => {
+        const key = listItem.nodeId.length > 0 ? listItem.nodeId : index;
```

Comment:
Goal of the change here was to use the ids of each node as the key for the list.
React wants unique keys to be used to help it know when to re-render things.

Since the list items here change often, using the index of the listItems isn't always ideal. In our case, we have placeholders which have no id set, so we just use the number (when user isn't in a selected tab).